### PR TITLE
APR-1362 - govuk frontend upgrade

### PR DESCRIPTION
**Pull request by gdbarnes**
This branch must be tested and deployed at the same time as:

`APR-1362_govuk_frontend_upgrade` on `das-assessor-service`

and

`APR-1362_govuk_frontend_upgrade` on `das-qna-api`

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml [hunk 1]
```diff
@@ -25,25 +25,23 @@
...
                                 isSamePage = pageTitle == pg.Title;
                                 pageTitle = pg.Title;
                                 nextPageTitle = "";
+
                                 if (i < pagesToCheck.Count - 1)
                                 {
                                     nextPageTitle = (pagesToCheck[i + 1]).Title;
                                 }
 
-                                @if (!isSamePage)
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml:42]: are you sure this should go?

File: src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml [hunk 2]
```diff
@@ -72,17 +71,19 @@
 
                                 @if (nextPageTitle != pageTitle)
                                 {
-                                    @Html.Raw("</ul>");
+                                    @Html.Raw("</ul></li>");
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml:74]: this </li> bit looks wrong to me

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml (hunk 2 above)
Comment (gdbarnes) [reply src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml:74]: I'm pretty sure that's right. I know what you're getting at, but it's closing a list item which contains an unordered list so it looks the wrong way round.

File: src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml [hunk 3]
```diff
@@ -25,25 +25,23 @@
             else
             {
                 <h1 class="govuk-heading-xl">@Model.Title</h1>
-                <ul class="govuk-list">
+
+                <ol class="app-task-list">
                     <li>
-                        <ul class="app-task-list govuk-!-margin-bottom-7">
+                        <h2 class="app-task-list__section">@Html.Raw(pagesToCheck[0].Title)</h2>
+                        <ul class="app-task-list__items app-task-list__items--no-number">
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml:32]: Have we checked all of this displays correctly as it's feels like very complex logic (which could probably benefit from being re-worked). From what I understand it's looking through all pages to check, and when it is a 'new page (i.e. a new title)' it dumps out the new title and puts stuff into a new unordered list 🤷‍♂️

Comment (gdbarnes) [reply src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml:31]: Seems to display correctly but I'm happy for the logic to be re-worked at a later date.

File: src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml (hunk 1 above)
Comment (gdbarnes) [reply src/SFA.DAS.ApplyService.Web/Views/Application/PagesWithSections.cshtml:42]: I've removed all reference to that now. I can't see any issue with the UI on first glance but lets see what comes out of testing.

**Review by MarkFCain — APPROVED** [review]

**Review by martin-hornsby — APPROVED** [review]

**Pull request merged**